math/complex_numbers.cpp

```diff
@@ -0,0 +1,249 @@
+/**
+ * @author tjgurwara99
+ * @file
+ *
+ * A basic implementation of Complex Number field as a class with operators
+ * overloaded to accommodate (mathematical) field operations.
+ */
+
+#include <cassert>
+#include <cmath>
+#include <complex>
+#include <ctime>
+#include <iostream>
+#include <stdexcept>
+
+/**
+ * Class Complex to represent complex numbers as a field.
+ */
+class Complex {
+    // The real value of the complex number
+    double re;
+    // The imaginary value of the complex number
+    double im;
+
+ public:
+    /**
+     * Complex Constructor which initialises the complex number which takes two
+     * arguments.
+     * @param x If the third parameter is 'true' then this x is the absolute
+     * value of the complex number, if the third parameter is 'false' then this
+     * x is the real value of the complex number (optional).
+     * @param y If the third parameter is 'true' then this y is the argument of
+     * the complex number, if the third parameter is 'false' then this y is the
+     * imaginary value of the complex number (optional).
+     * @param is_polar 'false' by default. If we want to initialise our complex
+     * number using polar form then set this to true, otherwise set it to false
+     * to use initialiser which initialises real and imaginary values using the
+     * first two parameters (optional).
+     */
+    explicit Complex(double x = 0.f, double y = 0.f, bool is_polar = false)
```

Comment:
agreed about the readability.
```suggestion
    explicit Complex(double x = 0.f, double y = 0.f, bool is_polar = false) {
       if (!polar) {
         re = x;
         im = y;
         return;
       }
       
       re = x * std::cos(y);
       im = x * std::sin(y);
   }
```